### PR TITLE
Improves documentation for NewTracer and NewTransport.

### LINKED ISSUE #65: tracer.NewNoopTracer support
Currently when writing a test that has a tracer as dependency, there is no easy way to create a `noop` tracer.

Ping @basvanbeek @adriancole

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR aims to improve documentation.

Ping @basvanbeek 

Closes #65

### CODE REVIEW

File: middleware/http/example_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		Transport: transport,
+	}
+
+	// creates the request
+	request, err := http.NewRequest("GET", "http://example.com", nil)
+	if err != nil {
+		log.Fatalf("unable to create the request: %+v\n", err)
+	}
+
+	// it is required to use the `Do` method and pass the request along
```
Comment (basvanbeek) [inline middleware/http/example_test.go]: this comment seems unfinished.

File: middleware/http/example_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	// as the request carries the context whereas in `Post`, `Get`, `Head`
+	c.Do(request.WithContext(ctx))
```
Comment (basvanbeek) [inline middleware/http/example_test.go]: I'd rather see you add an application span and add that to request Context. This way it is very unclear what ctx is actually doing.

File: example_test.go [hunk 3]
```diff
@@ -48,6 +49,16 @@ func ExampleNewTracer() {
 	// Output:
 }
 
+func ExampleNewTracer_tests() {
```
Comment (basvanbeek) [inline example_test.go]: what is the purpose of this function?

Comment (jcchavezs) [reply example_test.go]: The idea behind this was to have a new section in `godoc` for *New Tracer (tests)* showing how can you start a tracer to use in tests.

File: middleware/http/example_test.go (hunk 2 above)
Comment (jcchavezs) [reply middleware/http/example_test.go]: Makes totally sense. Will do so.

File: middleware/http/example_test.go (hunk 1 above)
Comment (jcchavezs) [reply middleware/http/example_test.go]: I am not sure, it continues in the next line:

> it is required to use the `Do` method and pass the request along as the request carries the context whereas in `Post`, `Get`, `Head`

Comment (basvanbeek) [reply middleware/http/example_test.go]: Yes which is an unfinished sentence. whereas in Post, Get, Head ... ???

https://www.merriam-webster.com/dictionary/whereas

File: example_test.go (hunk 3 above)
Comment (basvanbeek) [reply example_test.go]: Add the new section for godoc or remove this and take it to a next PR. Otherwise it dangles around without context.

**Comment by basvanbeek** [comment]
closing due to age and unresolved items

**Pull request closed without merging**